src/components/side-panel/layer-panel/vis-config-slider.js

```diff
@@ -47,34 +156,68 @@ export default function VisConfigSliderFactory(RangeSlider) {
     range,
     step,
     isRanged,
+    allowCustomValue,
     disabled,
     onChange,
     inputTheme
-  }) => (
-    <SidePanelSection disabled={Boolean(disabled)}>
-      {label ? (
-        <PanelLabel>
-          {typeof label === 'string' ? (
-            <FormattedMessage id={label} />
-          ) : typeof label === 'function' ? (
-            <FormattedMessage id={label(config)} />
-          ) : (
-            <FormattedMessage id={`property.${property}`} />
-          )}
-        </PanelLabel>
-      ) : null}
-      <RangeSlider
-        range={range}
-        value0={isRanged ? config.visConfig[property][0] : range[0]}
-        value1={isRanged ? config.visConfig[property][1] : config.visConfig[property]}
-        step={step}
-        isRanged={Boolean(isRanged)}
-        onChange={value => onChange({[property]: isRanged ? value : value[1]})}
-        inputTheme={inputTheme}
-        showInput
-      />
-    </SidePanelSection>
-  );
+  }) => {
+    const [custom, setCustom] = useState(false);
+    const value = config.visConfig[property];
+
+    const onChangeCheckbox = useCallback(() => {
+      if (custom) {
+        // we are swithcing from custom to not custom
+        // adjust value to range
+        const adjustedValue = isRanged
+          ? [clamp(range, value[0]), clamp(range, value[1])]
+          : clamp(range, value);
+        onChange({[property]: adjustedValue});
+      }
+      setCustom(!custom);
+    }, [onChange, property, isRanged, value, range, custom, setCustom]);
+
+    return (
+      <SidePanelSection disabled={Boolean(disabled)}>
+        {label ? (
+          <PanelLabel>
+            {typeof label === 'string' ? (
+              <FormattedMessage id={label} />
+            ) : typeof label === 'function' ? (
+              <FormattedMessage id={label(config)} />
+            ) : (
+              <FormattedMessage id={`property.${property}`} />
+            )}
+          </PanelLabel>
+        ) : null}
+
+        {allowCustomValue ? (
```

Comment:
It is also possible to have a saved maps with custom input value, so we need to add additional condition here to show CustomInputLabel if `props.value` is outside `props.range`. You can use `isInRange` from `filter-utils` to do the check

Reply:
> It is also possible to have a saved maps with custom input value, so we need to add additional condition here to show CustomInputLabel if `props.value` is outside `props.range`. You can use `isInRange` from `filter-utils` to do the check

{allowCustomValue && !isInRange(value, range) ? (
 did I understand correct?